From 54bde04591c75e7ac8e61f04f24e898e46b84464 Mon Sep 17 00:00:00 2001
From: Wendy Liang <wendy.liang@xilinx.com>
Date: Mon, 22 Jun 2020 15:24:24 -0700
Subject: [PATCH 1397/1851] dt-binding: soc: xilinx: ai-engine: Include SHIM
 row in partition

commit f0bf45dcde54738933c851803efe45ee99441d63 from
https://github.com/Xilinx/linux-xlnx.git

Specify clearly that the SHIM row needs to be included in the partition
"reg" property. The start row "0" is already shown in the example, need to
add the explanation.

Also fix the number of rows to include the SHIM row in the example.

Signed-off-by: Wendy Liang <wendy.liang@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/soc/xilinx/xlnx,ai-engine.yaml          | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml b/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml
index 581739f8688c..3292ac2ecd2a 100644
--- a/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml
+++ b/Documentation/devicetree/bindings/soc/xilinx/xlnx,ai-engine.yaml
@@ -60,9 +60,12 @@ patternProperties:
     properties:
       reg:
         description: |
-          It describes the group of tiles of the AI engine partition. The format
-          is defined by the parent AI engine device node's '#address-cells' and
-          '#size-cells' properties.
+          It describes the group of tiles of the AI engine partition. It needs
+          to include the SHIM row. The format is defined by the parent AI engine
+          device node's '#address-cells' and '#size-cells' properties. e.g. a v1
+          AI engine device has 2D tiles array, the first row is SHIM row. A
+          partition which has 50 columns and 8 rows of core tiles and 1 row of
+          SHIM tiles will be presented as <0 0 50 9>.
 
       label:
         maxItems: 1
@@ -86,7 +89,8 @@ examples:
       #size-cells = <2>;
 
       aie_partition0: aie-partition@0 {
-        reg = <0 0 50 8>;
+        /* 50 columns and 8 core tile rows + 1 SHIM row */
+        reg = <0 0 50 9>;
         xlnx,partition-id = <1>;
       };
     };
-- 
2.31.1

